utils/codepointtrie/src/codepointtrie.rs

```diff
@@ -673,6 +676,57 @@ where
     }
 }
 
+#[derive(PartialEq, Eq, Debug, Clone, Copy)]
+pub struct CodePointMapRange {
+    start: u32,
+    end: u32,
+    value: u32,
+}
```

Comment:
Nit: This should be parameterized by T:

```suggestion
pub struct CodePointMapRange<T: TrieValue> {
    start: u32,
    end: u32,
    value: T,
}
```

This allows us to keep the type safety information all the way through the iterator.